src/Tizen.Multimedia.Vision/MediaVision/InferenceFaceRecognier.cs

```diff
@@ -0,0 +1,175 @@
+        /// <feature>http://tizen.org/feature/vision.face_recognition</feature>
+        /// <since_tizen> 10 </since_tizen>
+        public InferenceFaceRecognizer()
+        {
+            ValidationUtil.ValidateFeatureSupported(VisionFeatures.FaceRecognition);
+
+            InteropFace.Create(out _handle).Validate("Failed to create face recognizer");
+
+            try
+            {
+                InteropFace.Prepare(_handle).Validate("Failed to prepare face recognizer");
+            }
+            catch (Exception e)
+            {
+                Log.Error(MediaVisionLog.Tag, e.ToString());
+                InteropFace.Destroy(_handle);
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the InferenceFaceRecognizer class.
+        /// </summary>
+        ~InferenceFaceRecognizer()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Registers face data to internal database.
+        /// </summary>
+        /// <param name="source">The face data to register.</param>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void RegisterFace(MediaVisionSource source, string label)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Register(_handle, source.Handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Unregisters face data from internal database.
+        /// </summary>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The InferenceFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void UnregisterFace(string label)
+        {
+            ValidateNotDisposed();
+
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Unregister(_handle, label).Validate("Failed to register face data");
```

Comment:
Unregister method could return some exception so how about adding try-catch statement like below?
 try
            {
               InteropFace.Unregister(_handle, label).Validate("Failed to register face data");
            }
            catch (Exception e)
            {
                Log.Error(MediaVisionLog.Tag, e.ToString());
                throw e;
             }

Reply:
I could be done with Validate() method.